src/test/java/wolox/training/testControllers/UserControllerTest.java

```diff
@@ -0,0 +1,149 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.util.NestedServletException;
+import wolox.training.controllers.UserController;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+
+@WebMvcTest(value = UserController.class, secure = false)
+@RunWith(SpringRunner.class)
+public class UserControllerTest {
+
+  @Autowired
+  private MockMvc mvc;
+
+  @MockBean
+  private UserRepository userRepository;
+
+  @MockBean
+  private BookRepository bookRepository;
+
+
+  @Test
+  public void getAllUSer_Test() throws Exception {
+
+    User User1 = new User();
+    User1.setUserName("user");
+    User1.setPassword("user");
+    User1.setName("user");
+    User1.setIsAdmin(true);
+    User1.setBirthDate(LocalDate.of(1995, 02, 11));
+
+    User User2 = new User();
+    User2.setUserName("user2");
+    User2.setPassword("user2");
+    User2.setIsAdmin(false);
+    User2.setName("user2");
+    User2.setBirthDate(LocalDate.of(1995, 02, 11));
+
+    List allUser = Arrays.asList(User1, User2);
+
+    given(userRepository.findAll()).willReturn(allUser);
+
+    mvc.perform(get("/api/users/allUsers")
+        .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$", hasSize(2)))
+        .andExpect(jsonPath("$[1].name", is(User2.getName())));
+  }
+
+
+  @Test(expected = RuntimeException.class)
+  public void deleteUser_Test() throws Throwable {
+    User User1 = new User();
+    User1.setUserName("user");
+    User1.setPassword("user");
+    User1.setName("user");
+    User1.setIsAdmin(true);
+    User1.setBirthDate(LocalDate.of(1995, 02, 11));
+
+    List allUser = Arrays.asList(User1);
+
+    given(userRepository.findAll()).willReturn(allUser);
+
+    mvc.perform(delete("/api/users/admin/delete/1")
+        .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk());
+    try {
+
+      mvc.perform(get("/api/users/findUser/1")
+          .contentType(MediaType.APPLICATION_JSON))
+          .andExpect(status().isNotFound())
+          .andExpect(jsonPath("$", hasSize(0)));
+    } catch (NestedServletException e) {
+      throw e.getCause();
+    }
+  }
+
+
+  @Test(expected = RuntimeException.class)
+  public void updateUser_Test() throws Throwable {
+    User User1 = new User();
+    User1.setUserName("user");
+    User1.setPassword("user");
+    User1.setName("user");
+    User1.setIsAdmin(true);
+    User1.setBirthDate(LocalDate.of(1995, 02, 11));
+
+    List allUser = Arrays.asList(User1);
+
+    given(userRepository.findAll()).willReturn(allUser);
+
+    String user =
+        "{\"id\": 1,\n\"userName\": \"user1\",\"password\": \"user1\",\"name\": \"User1\",\n"
```

Comment:
I don't like this, but i'm not sure if there's another way..if you want you can look into this